src/chess/bitboard.h

```diff
@@ -229,6 +238,86 @@ class BitBoard {
   std::uint64_t board_ = 0;
 };
 
+// Holds magic bitboard routines and performs initialization.
+// We use so-called "fancy" magic bitboards.
+class MagicBitBoards {
+ public:
+  // Constructor is called once in bitboard.cc to initialize structures, should
+  // not be called again!
+  MagicBitBoards();
```

Comment:
Actually what I wrote in the last paragraph I like more then singleton even..
But I wonder what's your opinion.

Reply:
I thought the current way of using a constructor was quite neat (I was a
bit in Java mindset where static initialization is very easy and common,
even without this constructor trick), but this fails miserably if the dummy
instance gets optimized away.

I like your last suggestion to move everything into anonymous namespace of
board.cc (dropping the class on top) and initialize explicitly. This has
the added advantage that we have full control where to do the
initialization (it will probably take only few milliseconds, but still...).
I'll implement the changes for this last approach in the coming days.

Reply:
I implemented the requested changes.

One additional benefit of the current approach is the fact that it's probably less sensitive to whether LTO is enabled, as all magic bitboard structures and code are now in the same compilation unit where they are used/called.